packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,315 @@
+import { EthExecutionAPI } from 'web3-common';
+import { Web3Context } from 'web3-core';
+import {
+	BlockTags,
+	convertToValidType,
+	HexString,
+	Numbers,
+	toHex,
+	ValidReturnTypes,
+	ValidTypes,
+} from 'web3-utils';
+import { privateKeyToAddress } from 'web3-eth-accounts';
+
+import Web3Eth from '.';
+import {
+	ChainIdMismatchError,
+	CommonOrChainAndHardforkError,
+	Eip1559GasPriceError,
+	Eip1559NotSupportedError,
+	InvalidGasOrGasPrice,
+	InvalidMaxPriorityFeePerGasOrMaxFeePerGas,
+	InvalidNonceOrChainIdError,
+	InvalidTransactionObjectError,
+	MissingChainOrHardforkError,
+	MissingCustomChainError,
+	MissingCustomChainIdError,
+	MissingGasError,
+	UnableToPopulateNonceError,
+	UnsupportedFeeMarketError,
+	UnsupportedTransactionTypeError,
+} from './errors';
+import { PopulatedUnsignedTransaction, Transaction } from './types';
+
+export function formatTransaction<
```

Comment:
For all these formatters, if we can define valid JSON-Schema (which I believe we can) then we can write a generic formatter which takes objects+schema+format and convert it. That can be more extensible approach.